packages/psss/src/containers/Tables/SiteReportTable.js

```diff
@@ -54,57 +55,59 @@ const TABLE_STATUSES = {
   LOADING: 'loading',
 };
 
-export const SiteReportTableComponent = React.memo(({ onSubmit, tableStatus, setTableStatus }) => {
+export const SiteReportTable = React.memo(({ tableStatus, setTableStatus, weekNumber }) => {
   const { fields } = useContext(EditableTableContext);
+  const { countryCode } = useParams();
 
-  const handleEdit = useCallback(() => {
-    setTableStatus(TABLE_STATUSES.EDITABLE);
-  }, [setTableStatus]);
-
-  const handleCancel = useCallback(() => {
-    setTableStatus(TABLE_STATUSES.STATIC);
-  }, [setTableStatus]);
-
-  const handleSubmit = async () => {
-    setTableStatus(TABLE_STATUSES.SAVING);
-    await onSubmit(fields);
-    setTableStatus(TABLE_STATUSES.STATIC);
-  };
+  const [saveReport] = useMutation(
+    () => {
+      setTableStatus(TABLE_STATUSES.SAVING);
+      FakeAPI.post(fields);
+    },
+    {
+      onSuccess: () => queryCache.invalidateQueries('country-weeks', { countryCode, weekNumber }),
+      onSettled: () => setTableStatus(TABLE_STATUSES.STATIC),
+    },
+  );
```

Comment:
I don't entirely agree. I think outside of the framework of redux, there isn't really the concept of "presentational" components anymore. I prefer to think of components as a piece of fully functional encapsulated and declarative ui. I don't really think that refetching data constitutes business logic and I think that other components using 'country-weeks' data are not impacted by refetching the data, they just display the latest available data from the server (which is the true centralised state). For point B, the api wont be a singleton any more after the api layer refactor.

Reply:
ps. It might be good to add the queryKey strings as constants.

Reply:
In saying that another option for abstracting some of this logic out side of the component, but without mapping the components together, would be to create a custom hook.

Reply:
Heh, I'm probably going to take a bit of convincing to come along for the ride on this.

What I mean about the 'country-weeks' is that you'd have situations where a certain component makes a change that invalidates data it doesn't fetch, but other components do. For example, the component responsible for updating a site's weekly case count of AFR would need to invalidate the 'country-weeks' endpoint too, even though it otherwise has no idea about country level weekly data. Understanding that seems well outside of the realm of the specific component.

On point B - I don't think it makes a difference if it's an api singleton or functions (potentially wrapped in an object), it still means that you have to do more funky stubbing than just injecting a prop if you want to be able to use/demo/test the component in isolation right?

That's about enough blood shed on this hill for me! I'm happy to be talked into the modern age if you're still keen on keeping fetching and query invalidation inside these components. Would love a third opinion from @mcccclean, though I think he's probably too busy with Tamanu atm.

Reply:
Your hook comment wasn't there when I wrote my reply - that seems a good modern pattern to pull out the bits I'm concerned about!